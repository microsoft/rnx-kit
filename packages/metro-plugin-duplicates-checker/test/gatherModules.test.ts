import {
  gatherModulesFromGraph,
  gatherModulesFromSourceMap,
  gatherModulesFromSources,
  normalizePath,
  resolveModule,
} from "../src/gatherModules";
import { bundleGraph, bundleSourceMap } from "./testData";

describe("normalizePath()", () => {
  test("trims Webpack URLs", () => {
    expect(normalizePath("webpack:///file")).toBe("file");
    expect(normalizePath("webpack:////file")).toBe("/file");
  });

  test("handles Windows paths", () => {
    expect(normalizePath("C:\\Users\\Arnold\\source\\rnx-kit")).toBe(
      "C:/Users/Arnold/source/rnx-kit"
    );
  });
});

describe("resolveModule()", () => {
  test("throws if a package is not found", () => {
    expect(() => resolveModule("/this-package-does-not-exist")).toThrow(
      "Unable to find package"
    );
  });
});

describe("gatherModulesFromGraph()", () => {
  test("builds module map from a basic source map", () => {
    const modules = gatherModulesFromGraph(bundleGraph, {});
    expect(Object.keys(modules).sort()).toMatchSnapshot();
  });
});

describe("gatherModulesFromSourceMap()", () => {
  test("builds module map from a basic source map", () => {
    const modules = gatherModulesFromSources(bundleSourceMap.sources, {});

    expect(Object.keys(modules).sort()).toMatchSnapshot();

    Object.keys(modules).forEach((name) => {
      expect(Object.keys(modules[name]).length).toBe(1);
    });

    expect(gatherModulesFromSourceMap(bundleSourceMap, {})).toEqual(modules);
  });
});
