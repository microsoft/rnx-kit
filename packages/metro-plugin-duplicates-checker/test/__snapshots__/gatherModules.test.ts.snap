// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatherModulesFromGraph() builds module map from a basic source map 1`] = `
Array [
  "fbjs",
  "react-native",
]
`;

exports[`gatherModulesFromSourceMap() builds module map from a basic source map 1`] = `
Array [
  "@babel/runtime",
  "abort-controller",
  "anser",
  "base64-js",
  "event-target-shim",
  "fbjs",
  "invariant",
  "metro",
  "nullthrows",
  "object-assign",
  "pretty-format",
  "promise",
  "prop-types",
  "react",
  "react-devtools-core",
  "react-is",
  "react-native",
  "react-refresh",
  "regenerator-runtime",
  "rnx-kit",
  "scheduler",
  "stacktrace-parser",
  "use-subscription",
  "whatwg-fetch",
]
`;
